logstash-core/lib/logstash/inputs/metrics.rb

```diff
@@ -44,9 +44,9 @@ def update(snapshot)
       #   - We can use a synchronization mechanism between the called thread (update method)
       #   and the plugin thread (run method)
       #   - How we handle back pressure here?
-      #   - one snashot should be only one event
-      snapshot.metric_store.to_events.each do |event|
-        @queue << event
+      #   - one snashot should be only one event ?
+      snapshot.metric_store.all_metrics.each do |metric|
```

Comment:
I am bit worried to use `each` in this case, since I don't want people thing its an enumerable when its not. I am OK to use `.all` like you suggested earlier?